modnet/models.py

```diff
@@ -554,79 +566,904 @@ class OR only return the most probable class.
+                    model_summary = ""
+                    for k in self.model[i].history.keys():
+                        model_summary += "{}: {:.4f}\t".format(k, self.model[i].history[k][-1])
+                    LOG.info(model_summary)
+        else:
+                ctx = multiprocessing.get_context('spawn')
+                pool = ctx.Pool(processes=n_jobs)
+                tasks =[]
+                for i,m in enumerate(self.model):
+                    m._make_picklable()
+                    tasks.append({'model':m, 'training_data':train_datas[i], 'model_id':i, **kwargs})
+                for res in tqdm.tqdm(pool.imap_unordered(_map_fit_MODNet, tasks, chunksize=1),
+                                     total=self.n_models):
+                    model, model_id = res
+                    model._restore_model()
+                    self.model[model_id] = model
+                    model_summary = f"Model #{model_id}\t"
+                    for k in model.history.keys():
+                        model_summary += "{}: {:.4f}\t".format(k, model.history[k][-1])
+                    LOG.info(model_summary)
+
+    def predict(self, test_data: MODData, return_unc=False, return_prob=False) -> pd.DataFrame:
+        """Predict the target values for the passed MODData.
+
+        Parameters:
+            test_data: A featurized and feature-selected `MODData`
+                object containing the descriptors used in training.
+            return_prob: For a classification tasks only: whether to return the probability of each
+                class OR only return the most probable class.
+            return_unc: wheter to return a second dataframe containing the uncertainties
+
+        Returns:
+            A `pandas.DataFrame` containing the predicted values of the targets.
+
+
+        """
+
+        all_predictions = []
+        for i in range(self.n_models):
+            p = self.model[i].predict(test_data, return_prob=return_prob)
+            all_predictions.append(p.values)
+
+        p_mean = np.array(all_predictions).mean(axis=0)
+        p_std = np.array(all_predictions).std(axis=0)
+
+        df_mean = pd.DataFrame(p_mean, index=p.index, columns=p.columns)
+        df_std = pd.DataFrame(p_std, index=p.index, columns=p.columns)
+
+        if return_unc:
+            return df_mean, df_std
+        else:
+            return df_mean
+
+    def evaluate(self, test_data: MODData) -> pd.DataFrame:
+        """Evaluates the target values for the passed MODData by returning the corresponding loss.
+
+        Parameters:
+            test_data: A featurized and feature-selected `MODData`
+                object containing the descriptors used in training.
+
+
+        Returns:
+            Loss score
+        """
+        all_losses = np.zeros(self.n_models)
+        for i,m in enumerate(self.model):
+            all_losses[i] = m.evaluate(test_data)
+
+        return all_losses.mean()
+
+
+    def fit_preset(
```

Comment:
This looks almost entirely repeated from the underlying `MODNetModel`, except at the end, and it is missing some changes you have made to fit_preset in this PR. I think we should rework this a bit